src/conversion/azure/network_security_converter.ts

```diff
@@ -43,20 +44,25 @@ function convertRule(
     source: 'data/azure/resource-graph-1.json',
   };
 
+  const constraints: Constraint = {};
   if (sourceIp) {
-    spec.sourceIp = sourceIp;
+    constraints.sourceIp = sourceIp;
   }
   if (sourcePort) {
-    spec.sourcePort = sourcePort;
+    constraints.sourcePort = sourcePort;
   }
   if (destinationIp) {
-    spec.destinationIp = destinationIp;
+    constraints.destinationIp = destinationIp;
   }
   if (destinationPort) {
-    spec.destinationPort = destinationPort;
+    constraints.destinationPort = destinationPort;
   }
   if (protocol) {
-    spec.protocol = protocol;
+    constraints.protocol = protocol;
+  }
```

Comment:
Could rewrite this section as such: 

const constraints: Constraint = {
    sourceIp,
    sourcePort,
    destinationIp,
    destinationPort,
    protocol,
  };

Reply:
I did it this way because I didn't want the protocol property to exist on the object if it was undefined. I could build the object as you say, and then use the new removeUndefinedProperties() function (in src/utilities) to strip off the fields that are undefined. With that said, I don't know if it matters anywhere that objects have properties with undefined values (vs omitting those properties).